server/src/weaverbird/backends/mongo_translator/steps/formula.py

```diff
@@ -0,0 +1,119 @@
+    else:
+        return '$' + name.id
+
+
+def mongo_formula_for_binop(binop: ast.BinOp, columns_aliases: Dict) -> dict:
+    if isinstance(binop.op, ast.Add):
+        mongo_op = '$add'
+    elif isinstance(binop.op, ast.Sub):
+        mongo_op = '$subtract'
+    elif isinstance(binop.op, ast.Mult):
+        mongo_op = '$multiply'
+    elif isinstance(binop.op, ast.Div):
+        mongo_op = '$divide'
+    elif isinstance(binop.op, ast.Pow):
+        mongo_op = '$pow'
+    else:
+        raise InvalidFormula(f'Operator {binop.op.__class__} is not supported')
+
+    translated_op = {
+        mongo_op: [
+            mongo_formula_for_ast_node(binop.left, columns_aliases),
+            mongo_formula_for_ast_node(binop.right, columns_aliases),
+        ]
+    }
+    #  In case of a division operation the translator must handle a 0 or null denominator.
+    #  The implemented logic is: if the translated operation is a division, wraps the operation with a
+    #  condition checking if the value of the right operand is in 0 or None and output a None result in this case
+    #  otherwise, return the result the division operation
+    if mongo_op == '$divide':
+        return {
+            '$cond': [
+                {'$in': [mongo_formula_for_ast_node(binop.right, columns_aliases), [0, None]]},
+                None,
+                translated_op,
+            ]
+        }
+    else:
+        return translated_op
+
+
+def mongo_formula_for_unaryop(unop: ast.UnaryOp, columns_aliases: Dict) -> dict:
+    if isinstance(unop.op, ast.USub):
+        return {'$multiply': [-1, mongo_formula_for_ast_node(unop.operand, columns_aliases)]}
+    else:
+        raise InvalidFormula(f'Operator {unop.op.__class__} is not supported')
+
+
+def mongo_formula_for_ast_node(node: ast.AST, columns_aliases: Dict) -> Any:
+    if isinstance(node, ast.BinOp):
+        return mongo_formula_for_binop(node, columns_aliases)
+    elif isinstance(node, ast.UnaryOp):
+        return mongo_formula_for_unaryop(node, columns_aliases)
+    elif isinstance(node, ast.Constant):
+        return mongo_formula_for_constant(node)
+    elif isinstance(node, ast.Name):
+        return mongo_formula_for_name(node, columns_aliases)
+    else:
+        raise InvalidFormula(f'Formula node {node} is not supported')
+
+
+def mongo_formula_for_expr(expr: ast.Expr, columns_aliases: Dict) -> dict:
+    if isinstance(expr.value, ast.AST):
+        return mongo_formula_for_ast_node(expr.value, columns_aliases)
+    else:
+        raise InvalidFormula
+
+
+def translate_formula(step: FormulaStep) -> list:
+    columns_aliases, sanitized_formula = sanitize_formula(step.formula)
+
+    module = ast.parse(sanitized_formula)
+    expr = module.body[0]
+    if not isinstance(expr, ast.Expr):
+        raise InvalidFormula
+
+    mongo_expr = mongo_formula_for_expr(expr, columns_aliases)
+    return [{'$addFields': {step.new_column: mongo_expr}}]
+
+
+def sanitize_formula(formula: str) -> Tuple[Dict, str]:
```

Comment:
```suggestion
def sanitize_formula(formula: str) -> Tuple[Dict[str, str], str]:
```
probably good to be as explicit as possible in other typings as well (`list`, `dict`, ...)

Reply:
:+1: